http/codegen/server.go

```diff
@@ -443,6 +444,27 @@ func {{ .RequestDecoder }}(mux goahttp.Muxer, decoder func(*http.Request) goahtt
 		{{- end }}
 	}
 
+	{{- else if .MapQueryParams }}
+	{
+		{{ .VarName }}Raw := r.URL.Query().Get("{{ .MapQueryParams }}")
+		{{- if .Required }}
+		if len({{ .VarName }}Raw) == 0 {
+			err = goa.MergeErrors(err, goa.MissingFieldError("{{ .Name }}", "query string"))
+		}
+		{{- end }}
+
+		{{- if not .Required }}
+		if len({{ .VarName }}Raw) != 0 {
+		{{- end }}
+			err2 := json.Unmarshal([]byte({{ .VarName }}Raw), &{{ .VarName }})
```

Comment:
how does this work? {{ .VarName }}Raw would be of type `url.Values` which is `map[string][]string` - how can this be cases to `[]byte` and then JSON unmarhsalled? In general I don't think we should use JSON to marhsal/unmarshal. The encoding / decoding code everywhere type asserts or type converts as needed and we should do that here too.

Reply:
`r.URL.Query().Get()` returns a string and it gets unmarshalled into `map[string][]string`.